File: services/overwatcher-frontend/src/types/event_log.ts
export interface EventLog {
  id: string;
  delivery_id: string;
  event_type: string;
  repo: string;
  sender: string;
  summary: string;
  created_at: string;
}

export interface EventLogListResponse {
  events: EventLog[];
  total: number;
  page: number;
  page_size: number;
}

File: services/overwatcher-backend/internal/api/http/dto/event_log.go
package dto

import "time"

type EventLogResponse struct {
	ID         string    `json:"id"`
	DeliveryID string    `json:"delivery_id"`
	EventType  string    `json:"event_type"`
	Repo       string    `json:"repo"`
	Sender     string    `json:"sender"`
	Summary    string    `json:"summary"`
	CreatedAt  time.Time `json:"created_at"`
}

type EventLogListResponse struct {
	Events   []EventLogResponse `json:"events"`
	Total    int64              `json:"total"`
	Page     int                `json:"page"`
	PageSize int                `json:"page_size"`
}
